infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -48,7 +48,7 @@ public class LogReplicationConfig {
     public static final int MAX_CACHE_NUM_ENTRIES = 200;
 
     // Percentage of log data per log replication message
-    public static final int DATA_FRACTION_PER_MSG = 90;
+    public static final long DATA_FRACTION_PER_MSG = 90L;
```

Comment:
I suppose changing the data-type here will not have any impact as the final variable is int.

Reply:
When it is used to compute the maxDataSizePerMsg `((maxMsgSize * DATA_FRACTION_PER_MSG) / 100L)` the multiplication before the division will happen with long division avoiding the overflow

Reply:
but is it required now with the max size being 25MB?

Reply:
yes even down to 25MB the value still comes right outside the int range